sdk/trace/batch_span_processor.go

```diff
@@ -166,32 +167,34 @@ func WithBlocking() BatchSpanProcessorOption {
 	}
 }
 
-func (bsp *BatchSpanProcessor) processQueue() {
-	batch := make([]*export.SpanData, 0, bsp.o.MaxExportBatchSize)
+// processQueue removes spans from the `queue` channel until there is
+// no more data.  It calls the exporter in batches of up to
+// MaxExportBatchSize until all the available data have been processed.
+func (bsp *BatchSpanProcessor) processQueue(batch *[]*export.SpanData) {
 	for {
-		var sd *export.SpanData
-		var ok bool
-		select {
-		case sd = <-bsp.queue:
-			if sd != nil && sd.SpanContext.IsSampled() {
-				batch = append(batch, sd)
+		// Read spans until either the buffer fills or the
+		// queue is empty.
+		ok := true
+		for ok && len(*batch) < bsp.o.MaxExportBatchSize {
+			var sd *export.SpanData
```

Comment:
Why did this declaration get moved inside the second `for` loop?

Reply:
Fixed.